Assets/Character/AaronTestAssets/IdleState.cs

```diff
@@ -0,0 +1,74 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class IdleState : PlayerState {
+
+    private Animator an;
+    private Player play;
+    private Rigidbody2D rgb;
+	void Start () {
+        play = GetComponent<Player>();
+        an = GetComponent<Animator>();
+        run = GetComponent<RunState>();
+        rgb = GetComponent<Rigidbody2D>();
+        roll = GetComponent<RollState>();
+        sr = GetComponent<SpriteRenderer>();
+        attack = GetComponent<AttackState>();
+        backstep = GetComponent<BackstepState>();
+        airborne = GetComponent<AirborneState>();
+        name = "Idle";
+	}
+	
+	// Update is called once per frame
+	public override void Update () {
+	}
+
+    public override void ComponentUpdate()
+    {
+        Debug.Log("IDLE");
+        if (!an.GetCurrentAnimatorStateInfo(0).IsName("Attack") && !an.GetCurrentAnimatorStateInfo(0).IsName("Attack")) //let attack animation finish
```

Comment:
I feeeel like attacking states are the only ones we'd wait for, but maybe putting all applicable states in an array and checking that instead would be better?